lib/x11utils.pm

```diff
@@ -213,7 +213,9 @@ sub handle_login {
     save_screenshot();
     # wait for DM, avoid screensaver and try to login
     # Previously this pressed esc, but that makes the text field in SDDM lose focus
-    send_key_until_needlematch('displaymanager', 'shift', 30, 3);
+    # we need send key 'esc' to quit screen saver when desktop is gnome
+    my $mykey = check_var('DESKTOP', 'gnome') ? 'esc' : 'shift';
+    send_key_until_needlematch('displaymanager', $mykey, 30, 3);
```

Comment:
If desktop is gnome, it will send 'esc' key until needlematch 'displaymanager'? Only send 'esc' key is OK? I think, If desktop is gnome, send 'esc' it quits gnome screen, then it should send 'shift' key to match 'displaymanger' needle.

Reply:
No, when desktop is gnome, it will make the focus point on displamanger after send key 'esc', no need to send key of shift.